enhancements/ibm-public-cloud/ibm-public-cloud-support.md

```diff
@@ -0,0 +1,243 @@
+---
+title: IBM Public Cloud Support
+authors:
+  - "@csrwng"
+reviewers:
+  - "@derekwaynecarr"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2020-02-04
+last-updated: 2020-02-04
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# IBM Public Cloud Support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+IBM Public Cloud deploys Kubernetes/OpenShift clusters by hosting their control plane
+as pods in a central management cluster. OpenShift v3.11 is currently offered as a managed
+service using this model. This type of setup allows the hosting of many
+cluster control planes on a single Kubernetes cluster. Only the worker nodes are exclusive
+to an individual user cluster and may exist in a different account than the central
+management cluster.
+
+Supporting this type of deployment in OpenShift 4.x requires a different approach to
+installing OpenShift and managing its control plane. Instead of bootstrapping a cluster 
+control plane from scratch on virtual machines, a cluster control plane can be created 
+by applying a set of manifests to an existing cluster in a new namespace. 
+
+Changes are required in different areas of the product in order to make clusters deployed 
+using this method viable. These include changes to the cluster version operator (CVO), web 
+console, second level operators (SLOs) deployed by the CVO, and RHCOS.
+
+## Motivation
+
+Enable OpenShift 4.x deployment on IBM Cloud.
+
+### Goals
+
+- Enable the deployment of OpenShift 4.x on IBM Public Cloud by providing the IBM
+  Public Cloud team the necessary tools to generate manifests needed for a hosted
+  control plane.
+- Ensure that this deployment model remains functional through regular e2e testing
+  on IBM Public Cloud.
```

Comment:
"regular" means we get a normal CI job in the openshift org making sure our control plane code changes don't break it?

Reply:
Will the CI job be blocking for OpenShift PRs? E.g. if deployment topology changes and some apiserver suddenly does not serve certain APIs because they moves, their CI will break.

Reply:
> "regular" means we get a normal CI job in the openshift org making sure our control plane code changes don't break it?

That is the plan. We currently have a periodic job that creates clusters based on the 4.3 branches. One will be added for the master/4.4 branch.


> Will the CI job be blocking for OpenShift PRs? E.g. if deployment topology changes and some apiserver suddenly does not serve certain APIs because they moves, their CI will break.

That will be harder. We would require capacity on the IBM Cloud to run that many jobs. Not sure that is feasible right now. The periodic job should block a release, but not individual PRs

Reply:
We describe those apiserver changes (they are happening, now for oauth) in enhancements. The IBM team has to watch that repo to be informed.